core/sdk-core/src/main/java/software/amazon/awssdk/core/SdkPojo.java

```diff
@@ -30,4 +30,18 @@
      * @return List of {@link SdkField} in this POJO. May be empty list but should never be null.
      */
     List<SdkField<?>> sdkFields();
+
+    /**
+     * Indicates whether some other object is "equal to" this one by sdk fields.
+     * This method only compares the non-inherited fields within this class.
+     *
+     * <p>
+     * If this class does not have a super class, this is essential same as {@link #equals(Object)}.
+     *
+     * @param other the object to be compared with
+     * @return true if the other object equals to this object by sdk fields, false otherwise.
+     */
+    default boolean equalsBySdkFields(Object other) {
```

Comment:
Should we clarify what is meant by "sdk fields". I know we use that phrasing elsewhere here, but it's not really made clear.

Reply:
Updated to 
```
Indicates whether some other object is "equal to" this one by SDK fields. 

An SDK field is a modeled, non-inherited field in an SdkPojo class, and is generated based on a service model.

If an SdkPojo class does not have any inherited fields, equalsBySdkFields and equals are essentially the same
```